gapic/src/main/com/google/gapid/models/Settings.java

```diff
@@ -224,10 +233,27 @@ private static int getInt(Properties properties, String name, int dflt) {
     }
   }
 
+  private static long getLong(Properties properties, String name, long dflt) {
+    String value = properties.getProperty(name);
+    if (value == null) {
+      return dflt;
+    }
+
+    try {
+      return Long.parseLong(value);
+    } catch (NumberFormatException e) {
+      return dflt;
+    }
+  }
+
   private static boolean getBoolean(Properties properties, String name) {
     return "true".equalsIgnoreCase(properties.getProperty(name, ""));
   }
 
+  private static boolean getBoolean(Properties properties, String name, boolean dflt) {
```

Comment:
This is unnecessary. If you want a boolean that is true by default, simply invert the meaning. I.e. rename autoCheckForUpdates to noAutoUpdates.